docs/StardustDocs/topics/operations/multiple/join.md

```diff
@@ -0,0 +1,393 @@
+
+Returns a new [`DataFrame`](DataFrame.md) that contains the merged rows and columns from both inputs.
+
+```kotlin
+join(otherDf, type = JoinType.Inner) [ { joinColumns } ]
+
+joinColumns: JoinDsl.(LeftDataFrame) -> Columns
+
+interface JoinDsl: LeftDataFrame {
+    
+    val right: RightDataFrame
+    
+    fun DataColumn.match(rightColumn: DataColumn)
+}
+```
+
+`joinColumns` is a [column selector](ColumnSelectors.md) that defines column mapping for join:
+
+Related operations: [](multipleDataFrames.md)
+
+## Examples
+
+### Join with explicit keys (with different names) {collapsible="true"}
+
+Use the Join DSL when the key column names differ:
+
+- access the right `DataFrame` via `right`;
+- define the join condition with **match()**.
+
+<!---FUN notebook_test_join_3-->
+
+```kotlin
+dfAges
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_3.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_5-->
+
+```kotlin
+dfCities
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_5.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_6-->
+
+```kotlin
+// INNER JOIN on differently named keys:
+// Merge a row when dfAges.firstName == dfCities.name.
+// With the given data all 3 names match → all rows merge.
+dfAges.join(dfCities) { firstName match right.name }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_6.html" width="100%" height="500px"></inline-frame>
+
+### Join with explicit keys (with the same names) {collapsible="true"}
+
+If mapped columns have the same name, just select join columns (one or several) from the left [`DataFrame`](DataFrame.md):
+
+<!---FUN notebook_test_join_8-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_8.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_10-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_10.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_11-->
+
+```kotlin
+// INNER JOIN on "name" only:
+// Merge when left.name == right.name.
+// Duplicate keys produce multiple merged rows (one per pairing).
+dfLeft.join(dfRight) { name }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_11.html" width="100%" height="500px"></inline-frame>
+
+> In this example, the "city" columns from the left and right dataframes do not match to each other.
+> After joining, the "city" column from the right dataframe is included in the result dataframe
+> with the name **"city1"** to avoid a name conflict.  
+> { style = "note" }
+
+### Join with implicit keys (all columns with the same name) {collapsible="true"}
+
+If `joinColumns` is not specified, columns with the same name from both [`DataFrame`](DataFrame.md)
```

Comment:
You say `joinColumns` is not specified, yet in the example, you show `{ name and city }`